File: C  programs/Problems on Pattern Printing/String Pattern/Half Pyramid.c
/* Write a program which accept String from user and display below pattern.

Input : Hello
Output :  H  
          H  e  
		  H  e  l  
		  H  e  l  l       
		  H  e  l  l  o      

Input : "Marvellous"
Output : M 
         M a
		 M a r
		 M a r v 
		 M a r v e
		 M a r v e l 
		 M a r v e l l 
		 M a r v e l l o 
		 M a r v e l l o u 
		 M a r v e l l o u s

*/    

#include<stdio.h>

void Pattern(char str[])
{
	int i=0;
	int j=0;
	int k=0;
	int iCnt=0;
	int iLength=0;
	int iCol=1;

	while(str[iLength]!='\0')
	{
		iLength++;
		iCnt++;
	}

    iCol=iLength;
	for(i=1;i<=iLength;i++)
	{
		k=0;

		for(j=1;j<=iCol;j++)
		{
			printf("%c\t",str[k]);
			k++;
		}
		iCol++;
		printf("\n");
	}

}

int main()
{
	char str[30]={'\0'};

	printf("Enter string: \n");
	scanf("%[^'\n']s",&str);

	Pattern(str);

	return 0;
}
		   

File: Java programs/Problems on Pattern Printing/String Pattern/Half pyramid.java
/* Write a java program which accept String from user and display below pattern.

Input : Hello
Output :  H  
          H  e  
		  H  e  l  
		  H  e  l  l       
		  H  e  l  l  o      

*/    

import java.util.*;

class Pattern
{
    public void Pattern(String str)
    {
	    char Arr[]=str.toCharArray();
	    int i=0,j=0;
       	    
        for(i=0;i<Arr.length;i++)        
	    {
		  for(j=0;j<Arr.length;j++)
		  {
			 if(i>=j)
			 {
			   System.out.print(Arr[j]+"\t");
			 }
		  }	
		  System.out.println();
	    }
    }
}

class Lba383
{
	public static void main(String arg[])
	{
		Scanner sobj=new Scanner(System.in);
		
		System.out.println("Enter string");
		String str=sobj.nextLine();
		
		Pattern pobj=new Pattern();
		pobj.Pattern(str);
	}	
}

